presto-jdbc/src/main/java/com/facebook/presto/jdbc/QueryExecutor.java

```diff
@@ -50,20 +50,16 @@
     private final JsonCodec<ServerInfo> serverInfoCodec;
     private final HttpClient httpClient;
 
-    private QueryExecutor(String userAgent, JsonCodec<QueryResults> queryResultsCodec, JsonCodec<ServerInfo> serverInfoCodec, HostAndPort socksProxy)
+    private QueryExecutor(String userAgent, JsonCodec<QueryResults> queryResultsCodec, JsonCodec<ServerInfo> serverInfoCodec, HttpClient httpClient)
     {
         requireNonNull(userAgent, "userAgent is null");
         requireNonNull(queryResultsCodec, "queryResultsCodec is null");
         requireNonNull(serverInfoCodec, "serverInfoCodec is null");
+        requireNonNull(httpClient, "httpClient is null");
```

Comment:
Move this check to the assignment and fix the other two also. Then this constructor will just be a few simple assignments.